app/service/data_svc.py

```diff
@@ -157,6 +178,9 @@ def __init__(self):
         phase_id = 0
         while phase_id < len(phases):
             for idx, step in enumerate(phases[phase_id]):
+                if isinstance(step, Ability):
```

Comment:
I usually hate instance checks. this is a good use though.

Reply:
yea, I wasn't too keen on it either but we were getting mixed types as things merge.

This also would probably be part of how we handle adversary loading in general that might be a good place to go over and maybe simplify as we treat an ability with an id of an adversary as an adversary pack that is offset in phases (happens lower down in the 'else' clause).